File: ir/exprstmt_ir.py
from common.span import SourceSpan
from .expr_ir import ExprIR
from .stmt_ir import StmtIR


class ExprStmtIR(StmtIR):
    def __init__(self, value: ExprIR, span: SourceSpan):
        super().__init__(span=span)
        self.span=span
        self.value=value

File: rust/src/ir/nodes/exprstmt_ir.rs
use crate::ir::{ExprIR, SourceSpan};

#[derive(Debug, Clone)]
pub struct ExprStmtIR {
    pub value: Option<Box<ExprIR>>,
    pub span: Option<SourceSpan>,
}

impl ExprStmtIR {
    pub fn new(value: Option<ExprIR>, span: Option<SourceSpan>) -> Self {
        Self {
            value: value.map(Box::new),
            span,
        }
    }
}
